### PR TITLE
HttpClient getHeader can return empty string

### LINKED ISSUE #11999: TypeError: Return value of OC\Http\Client\Response::getHeader() must be of the type string, null returned
It appears that [OC\Http\Client\Response](https://github.com/nextcloud/server/blob/master/lib/private/Http/Client/Response.php#L73) sometimes returns `null`, when it's expected it shouldn't. This happens in the bookmarks app e.g. when adding a bookmark. This is because guzzle6 [returns an empty array](http://docs.guzzlephp.org/en/stable/psr7.html#headers) in case the requested header doesn't exist.

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #11999

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: lib/private/Http/Client/Response.php [hunk 1]
```diff
@@ -70,8 +70,14 @@ public function getStatusCode(): int {
 	 * @param string $key
 	 * @return string
 	 */
-	public function getHeader(string $key): string {
-		return $this->response->getHeader($key)[0];
+	public function gectHeader(string $key): string {
```
Comment (danielkesselberg) [inline lib/private/Http/Client/Response.php]: :thinking:

Comment (rullzer) [reply lib/private/Http/Client/Response.php]: Yeah that didn't go as planned

**Review by ChristophWurst — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]
Makes sense 👍

**Comment by MorrisJobke** [comment]
CI fails:

```
PHP Fatal error:  Class OC\Http\Client\Response contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (OCP\Http\Client\IResponse::getHeader) in /drone/src/github.com/nextcloud/server/lib/private/Http/Client/Response.php on line 35
```

**Comment by MorrisJobke** [comment]
@rullzer Is a backport needed?

**Comment by rullzer** [comment]
Yeah since it an otherwise return null which php doesn't like in 7 etc ;)

**Comment by MorrisJobke** [comment]
> The backport to stable13 failed. Please do this backport manually.

@rullzer Are you doing this?

**Comment by MorrisJobke** [comment]
> @rullzer Are you doing this?

❓

**Comment by rullzer** [comment]
Nope. We had a different guzzle on 13. So no need as far as I can tell.

**Pull request merged**